src/herder/test/UpgradesTests.cpp

```diff
@@ -553,6 +624,164 @@ TEST_CASE("Ledger Manager applies upgrades properly", "[upgrades]")
+                                    makeConfigUpgrade(*configUpgradeSetFrame)),
+                                outUpgrade, *app, ltx, header) ==
+                            Upgrades::UpgradeValidity::XDR_INVALID);
+                };
+                SECTION("no updated entries")
+                {
+                    testInvalidXdr();
+                }
+                SECTION("duplicate entries")
+                {
+                    badConfigUpgradeSet.updatedEntry.emplace_back(
+                        CONFIG_SETTING_CONTRACT_MAX_SIZE_BYTES);
+                    badConfigUpgradeSet.updatedEntry.emplace_back(
+                        CONFIG_SETTING_CONTRACT_MAX_SIZE_BYTES);
+                    testInvalidXdr();
+                }
+            }
+        }
+        SECTION("bad value")
+        {
+            REQUIRE(Upgrades::isValidForApply(
+                        toUpgradeType(makeConfigUpgrade(
+                            *makeMaxContractSizeBytesTestUpgrade(ltx, 0))),
+                        outUpgrade, *app, ltx,
+                        header) == Upgrades::UpgradeValidity::INVALID);
+        }
+    }
+
+    SECTION("validate for nomination")
+    {
+        LedgerUpgradeType outUpgradeType;
+        SECTION("valid")
+        {
+            REQUIRE(Upgrades(scheduledUpgrades)
+                        .isValid(
+                            toUpgradeType(makeConfigUpgrade(*configUpgradeSet)),
+                            outUpgradeType, true, *app, ltx, header));
+        }
+        SECTION("not valid")
+        {
+            SECTION("no upgrade scheduled")
+            {
+                REQUIRE(!Upgrades().isValid(
+                    toUpgradeType(makeConfigUpgrade(*configUpgradeSet)),
+                    outUpgradeType, true, *app, ltx, header));
+            }
+            SECTION("inconsistent value")
+            {
+                REQUIRE(!Upgrades(scheduledUpgrades)
+                             .isValid(toUpgradeType(makeConfigUpgrade(
+                                          *makeMaxContractSizeBytesTestUpgrade(
+                                              ltx, 12345))),
+                                      outUpgradeType, true, *app, ltx, header));
+            }
+        }
+    }
+}
+
+TEST_CASE("config upgrades applied to ledger", "[upgrades]")
+{
+    VirtualClock clock;
+    auto cfg = getTestConfig(0);
+    cfg.TESTING_UPGRADE_LEDGER_PROTOCOL_VERSION =
+        static_cast<uint32_t>(SOROBAN_PROTOCOL_VERSION) - 1;
+    auto app = createTestApplication(clock, cfg);
+
+    // Need to actually execute the upgrade to v20 to get the config
+    // entries initialized.
+    executeUpgrade(*app, makeProtocolVersionUpgrade(
+                             static_cast<uint32_t>(SOROBAN_PROTOCOL_VERSION)));
+
+    SECTION("unknown config upgrade set results in exception")
+    {
+        // TODO: Use function
+        auto contractID = autocheck::generator<Hash>()(5);
```

Comment:
nit: Please remove 5 here. These are pretty prevalent in the code and they do literally nothing. I think we could avoid this at least in the new code.

Reply:
The `()` operator requires a size parameter. Is there a way around this that I'm unaware of?

Reply:
Sorry, I confused this with a set of generator functions that have a default value. I don't think the size is being used here, but probably the generator function itself doesn't provide a default value, so I guess we have to keep this.